Runtime/Rust/src/rpc/mod.rs

```diff
@@ -0,0 +1,79 @@
+//! Runtime code to support Bebop RPC. Checkout https://github.com/RainwayApp/bebop/wiki/rpc for
+//! more information.
+
+use std::num::NonZeroU16;
+use std::time::Duration;
+
+pub use bebop_handler_macro::handlers;
+pub use datagram::{
+    owned::RpcDatagram as OwnedDatagram,
+    owned::{
+        RpcServiceNameArgs as OwnedRpcServiceNameArgs,
+        RpcServiceNameReturn as OwnedRpcServiceNameReturn,
+    },
+    RpcDatagram as Datagram, RpcRequestHeader as RequestHeader,
+    RpcResponseHeader as ResponseHeader, RpcServiceNameArgs, RpcServiceNameReturn,
+};
+pub use datagram_info::DatagramInfo;
+pub use deadlines::Deadline;
+pub use error::*;
+pub use router::*;
+pub use transport::TransportProtocol;
+
+pub type DynFuture<'a, T = ()> =
+    std::pin::Pin<Box<dyn 'a + Send + std::future::Future<Output = T>>>;
+
+mod datagram;
+mod datagram_info;
+mod deadlines;
+mod error;
+mod router;
+mod transport;
+
+#[inline]
+pub fn convert_timeout(timeout: Option<Duration>) -> Option<NonZeroU16> {
+    timeout.and_then(|t| {
+        let t = t.as_secs();
+        debug_assert!(t < u16::MAX as u64, "Maximum timeout is 2^16-1 seconds.");
```

Comment:
Should this be a user-facing error rather than just an assertion? A user might not know about this restriction and define an RPC with a large timeout, right?

Reply:
Maybe? I was thinking it would usually be something they write into the code so it would be a code bug.